src/components/PlaylistModal/PlaylistOption/index.jsx

```diff
@@ -0,0 +1,99 @@
+import React from "react";
+import { useAuth, useUserData } from "contexts";
+import {
+	deletePlaylistService,
+	deleteVideoFromPlaylistService,
+	postVideoToPlaylist,
+} from "services";
+import { useToast } from "custom-hooks/useToast";
+import { Delete } from "@mui/icons-material";
+
+const PlaylistOption = ({ video, playlist }) => {
+	const { userDataDispatch } = useUserData();
+	const { showToast } = useToast();
+	const { authToken } = useAuth();
+
+	const isVideoInPlaylist =
+		playlist.videos.find(
+			(playlistVideo) => playlistVideo._id === video._id
+		) === undefined
+			? false
+			: true;
+
+	const handleChangePlaylist = async (e) => {
+		try {
+			const {
+				data: { playlist: updatedPlaylist },
+			} = isVideoInPlaylist
+				? await deleteVideoFromPlaylistService(
+						authToken,
+						playlist._id,
+						video._id
+				  )
+				: await postVideoToPlaylist(authToken, playlist._id, video);
+			userDataDispatch({
+				type: "UPDATE_PLAYLISTS",
+				payload: { playlist: updatedPlaylist },
+			});
+			showToast(
+				isVideoInPlaylist
+					? "Video removed from playlist."
+					: "Video added to playlist.",
+				"success"
+			);
+		} catch (error) {
+			showToast(
+				isVideoInPlaylist
+					? "Failed to remove video playlist."
+					: "Failed to add video to playlist.",
+				"error"
+			);
+		}
+	};
+
+	const handleDeletePlaylist = async (e) => {
+		try {
+			const {
+				data: { playlists },
+			} = await deletePlaylistService(authToken, playlist._id);
+			userDataDispatch({
+				type: "SET_PLAYLISTS",
```

Comment:
suggestion: You can store all type values for reducer in one file as constants, It will helpful when you need to change any constants.

Reply:
Sure, thank you for this suggestion.